pkg/resources/tag_association_acceptance_test.go

```diff
@@ -1,151 +1,283 @@
+					testAccCheckTableColumnTagAssociation(tagID, objectID2, "v1"),
+				),
+			},
+		},
+	})
 }
 
-resource "snowflake_table" "test_table" {
-	name                = "%[1]v"
-	database            = "%[3]v"
-	schema              = "%[4]v"
-
-	column {
-		name    = "column_name"
-		type    = "VARIANT"
+func testAccCheckTableColumnTagAssociation(tagID sdk.SchemaObjectIdentifier, objectID sdk.ObjectIdentifier, tagValue string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		db := acc.TestAccProvider.Meta().(*sql.DB)
+		client := sdk.NewClientFromDB(db)
+		ctx := context.Background()
+		tv, err := client.SystemFunctions.GetTag(ctx, tagID, objectID, sdk.ObjectTypeColumn)
+		if err != nil {
+			return err
+		}
+		if tagValue != tv {
+			return fmt.Errorf("expected tag value %s, got %s", tagValue, tv)
+		}
+		return nil
 	}
 }
 
-resource "snowflake_tag_association" "columnTag" {
-	object_identifier {
-		database            = "%[3]v"
-		schema              = "%[4]v"
-		name     = "${snowflake_table.test_table.name}.${snowflake_table.test_table.column[0].name}"
+func TestAcc_TagAssociationAccountIssues1910(t *testing.T) {
+	// todo: use role with ORGADMIN in CI (SNOW-1165821)
+	// SNOWFLAKE_TEST_ACCOUNT_CREATE must be set to 1 to run this test
+	if _, ok := os.LookupEnv("SNOWFLAKE_TEST_ACCOUNT_CREATE"); !ok {
+		t.Skip("Skipping TestInt_AccountCreate")
+	}
+	tagName := "tag-" + strings.ToUpper(acctest.RandStringFromCharSet(4, acctest.CharSetAlpha))
+	accountName := "account_" + strings.ToUpper(acctest.RandStringFromCharSet(4, acctest.CharSetAlpha))
+	resourceName := "snowflake_tag_association.test"
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"tag_name":     config.StringVariable(tagName),
+			"account_name": config.StringVariable(accountName),
+			"database":     config.StringVariable(acc.TestDatabaseName),
+			"schema":       config.StringVariable(acc.TestSchemaName),
+		}
 	}
 
-	object_type = "COLUMN"
-	tag_id      = snowflake_tag.tag1.id
-	tag_value   = "TAG_VALUE"
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_TagAssociation/issue1910"),
+				ConfigVariables: m(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "object_type", "ACCOUNT"),
+					resource.TestCheckResourceAttr(resourceName, "tag_id", fmt.Sprintf("%s|%s|%s", acc.TestDatabaseName, acc.TestSchemaName, tagName)),
+					resource.TestCheckResourceAttr(resourceName, "tag_value", "v1"),
+				),
+			},
+		},
+	})
 }
-`, n1, n2, databaseName, schemaName)
+
+func TestAcc_TagAssociationIssue1926(t *testing.T) {
```

Comment:
How it will behave if we set the table name to "table.test" and column to "column" or column to "test.column"?

Reply:
Get a lot of issues with Table resource. It simply cannot handle identifiers or columns with these names. Has nothing to do with tag or tag association. First I just added the 

```
query ALTER TABLE "terraform_test_database"."terraform_test_schema"."table" MODIFY COLUMN "test.test.column" SET TAG "terraform_test_database"."terraform_test_schema"."tag-XLKK" = 'v1' err 002003 (42S02): SQL compilation error:
Table '"terraform_test_database"."terraform_test_schema"."table"' does not exist or not authorized.
```

and also:
```
2024/03/04 22:54:18 [DEBUG] sql-conn-exec: [query ALTER TABLE "terraform_test_database"."terraform_test_schema"."table.test" DROP COLUMN "test.column" err 001432 (42601): SQL compilation error:
cannot drop all the columns of a table duration 109.416409ms args {}] ()
```